src/item.cpp

```diff
@@ -5512,30 +5600,38 @@ bool item::is_power_armor() const
     return t->power_armor;
 }
 
-int item::get_encumber( const Character &p, encumber_flags flags ) const
+int item::get_encumber( const Character &p, const bodypart_id &bodypart,
+                        encumber_flags flags ) const
 {
     const islot_armor *t = find_armor_data();
     if( t == nullptr ) {
         // handle wearable guns (e.g. shoulder strap) as special case
         return is_gun() ? volume() / 750_ml : 0;
     }
-    int encumber = t->encumber;
+
+    int encumber = 0;
 
     // Additional encumbrance from non-rigid pockets
     float relative_encumbrance = 1;
     if( !( flags & encumber_flags::assume_full ) ) {
         relative_encumbrance = contents.relative_encumbrance();
     }
 
-    encumber += std::ceil( relative_encumbrance * ( t->max_encumber - t->encumber ) );
+    for( const auto &data : t->data ) {
+        if( bodypart == bodypart_id( bodypart_str_id( "num_bp" ) ) ||
+            ( data.covers.has_value() && data.covers.value().test( bodypart.id() ) ) ) {
```

Comment:
```suggestion
            ( data.covers && data.covers->test( bodypart.id() ) ) ) {
```
Accessing a `optional<T>` this way is shorter and that style is used in most of the code.